packages/core/documentation/src/components/ProductDocumentModal.vue

```diff
@@ -0,0 +1,398 @@
+            </KExternalLink>
+          </div>
+        </div>
+      </template>
+      <template #body-content>
+        <KAlert
+          v-if="errorMessage"
+          :alert-message="errorMessage"
+          appearance="danger"
+          class="bottom-spacing"
+        />
+
+        <!-- TODO: help tooltip-->
+        <KFileUpload
+          :accept="['.md', '.markdown']"
+          :button-text="fileUploadButtonText"
+          class="document-file-upload"
+          data-testid="document-file-upload"
+          :label="i18n.t('documentation.form_modal.file_label')"
+          :label-attributes="{ help: i18n.t('documentation.form_modal.file_tooltip') }"
+          :placeholder="isEditing ? filePlaceholderText : undefined"
+          @file-added="handleFileSelected"
+          @file-removed="handleFileRemoved"
+        />
+
+        <div class="document-inputs">
+          <div class="page-name side-by-side">
+            <KInput
+              v-model.trim="formData.pageName"
+              data-testid="documentation-page-name"
+              :label="i18n.t('documentation.form_modal.title_label')"
+              :placeholder="i18n.t('documentation.form_modal.title_placeholder')"
+              required
+            />
+          </div>
+          <div class="url-slug side-by-side">
+            <KInput
+              v-model.trim="formData.urlSlug"
+              data-testid="documentation-url-slug"
+              :error-message="i18n.t('documentation.form_modal.slug_error')"
+              :has-error="slugError"
+              :label="i18n.t('documentation.form_modal.slug_label')"
+              :pattern="slugRegex"
+              :placeholder="i18n.t('documentation.form_modal.slug_placeholder')"
+            />
+          </div>
+        </div>
+
+        <div
+          v-if="!hidePublishToggle"
+          class="documentation-status"
+        >
+          <div>
+            <KLabel>
+              {{ i18n.t('documentation.form_modal.status_label') }}
+            </KLabel>
+          </div>
+          <KInputSwitch
+            v-model="publishModel"
+            data-testid="documentation-status"
+            :label="publishedStatusText"
+          />
+        </div>
+
+        <KSelect
+          v-model="formData.parent"
+          appearance="select"
+          class="document-parent-select"
+          data-testid="documentation-parent"
+          enable-filtering
+          :items="availableParentDocuments"
+          :label="i18n.t('documentation.form_modal.parent_document_select_label')"
+          width="100%"
+        />
+      </template>
+
+      <template #action-buttons>
+        <div class="action-buttons">
+          <KButton
+            v-if="isEditing"
+            appearance="danger"
+            class="edit-documentation-delete-button"
+            data-testid="edit-documentation-delete-button"
+            :disabled="state.matches('pending')"
+            :icon="state.matches('pending') ? 'spinner' : undefined"
```

Comment:
Same comment here on not using the icon prop; all occrences

Reply:
fixed